### PR TITLE
Fix ord functor

### PR DESCRIPTION
**Pull request by b-scholz**
This is related to issue #940. 

The ord() functor permits any type of argument with this pull-request.

### CODE REVIEW

File: src/AstTypeAnalysis.cpp [hunk 1]
```diff
@@ -437,13 +437,15 @@ std::map<const AstArgument*, TypeSet> TypeAnalysis::analyseTypes(
...
 
             // add a constraint for each argument of the functor
-            for (size_t i = 0; i < fun.getArity(); i++) {
-                auto arg = getVar(fun.getArg(i));
-                if (fun.acceptsNumbers(i)) {
-                    addConstraint(isSubtypeOf(arg, env.getNumberType()));
-                }
-                if (fun.acceptsSymbols(i)) {
-                    addConstraint(isSubtypeOf(arg, env.getSymbolType()));
+            if (fun.getFunction() != FunctorOp::ORD) {
```
Comment (mmcgr) [inline src/AstTypeAnalysis.cpp]: Could you write this as
```
 if (fun.getFunction() == FunctorOp::ORD) {
    return;
}
```

File: src/AstSemanticChecker.cpp [hunk 2]
```diff
@@ -216,13 +216,15 @@ void AstSemanticChecker::checkProgram(ErrorReport& report, const AstProgram& pro
...
 
         // check types of arguments
-        for (size_t i = 0; i < fun.getArity(); i++) {
-            auto arg = fun.getArg(i);
-            if (fun.acceptsNumbers(i) && !isNumberType(typeAnalysis.getTypes(arg))) {
-                report.addError("Non-numeric argument for functor", arg->getSrcLoc());
-            }
-            if (fun.acceptsSymbols(i) && !isSymbolType(typeAnalysis.getTypes(arg))) {
-                report.addError("Non-symbolic argument for functor", arg->getSrcLoc());
+        if (fun.getFunction() != FunctorOp::ORD) {
```
Comment (mmcgr) [inline src/AstSemanticChecker.cpp]: Could you write this as
```
 if (fun.getFunction() == FunctorOp::ORD) {
    return;
}
```
The for loop handling the normal case is then unchanged.

File: src/AstTypeAnalysis.cpp (hunk 1 above)
Comment (b-scholz) [reply src/AstTypeAnalysis.cpp]: Will do.

File: src/AstSemanticChecker.cpp (hunk 2 above)
Comment (b-scholz) [reply src/AstSemanticChecker.cpp]: Will do.

**Pull request merged**